backend/app/categories/controller.py

```diff
@@ -38,8 +39,17 @@ async def get_category_route(category_id: CategoryEnum,
 async def get_category_item_route(category_id: CategoryEnum, item_id: ObjectID,
                                   db: AsyncIOMotorClient = Depends(get_database)) -> ItemInResponse:
     """Get the details about a particular item"""
-    # TODO: add particular item fetching
-    return {"success": True}
+
+    if (ObjectId.is_valid(item_id)==False):
+        """ Validation for item_id """
+        raise HTTPException(
+            status_code=400, detail='Not a valid ObjectID')
```

Comment:
This check is not required, `item_id: ObjectID` already verifies whether it is a valid ObjectId

Reply:
The server error has been fixed with the latest PR.